solidity/contracts/PhasedEscrow.sol

```diff
@@ -47,6 +49,13 @@ contract PhasedEscrow is Ownable {
 
         beneficiary.__escrowSentTokens(amount);
     }
+
+    /// @notice Withdraws all funds from a non-phased Escrow passed as
+    ///         a parameter. For this function to succeed, this PhasedEscrow
+    ///         has to be set as a beneficiary of the non-phased Escrow.
+    function withdrawFromEscrow(Escrow _escrow) public {
```

Comment:
I take that we don't care if a non-owner calls this functions since the address of beneficiary will be set by the owner?

Reply:
Yes, [Escrow is Ownable](https://github.com/keep-network/keep-core/blob/solidity/v1.4.0/solidity/contracts/Escrow.sol) and the owner has to first set beneficiary via `setBeneficiary()` and then, the beneficiary can withdraw via `withdraw()`. I don't see a threat with this function being public as a non-phased `Escrow` can only drain funds to 0, we can not withdraw funds in portions.